kernel/include/sentry/zlib/crypto.h

```diff
@@ -4,20 +4,19 @@
 #ifndef ZLIB_CRYPTO_H
 #define ZLIB_CRYPTO_H
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <inttypes.h>
 
 /** \addtogroup CRC
  *  @{
  */
 
-/*
-    requires \valid(buf + (0 .. (len-1)));
-    assigns buf[0 .. (len -1)];
-   */
+/*@
+   requires len >= 0;
+   requires buf != \null ==> \valid_read(buf + (0 .. len - 1));
```

Comment:
does this require clause valid if `buf != null` **and** `len == 0`

Reply:
fixed